filament/src/components/RenderableManager.cpp

```diff
@@ -204,83 +207,30 @@ RenderableManager::Builder& RenderableManager::Builder::enableSkinningBuffers(bo
-        "bone indices and weights pairs count (%u) must be a multiple of vertex counts of all primitives (%u)",
-        count, allVerticesCount);
-    size_t pairSkipped = 0;
-    for (size_t iEntry = 0; iEntry < primitiveCount; iEntry++){
-        size_t  pairsPerPrimitiveCount = pairsPerAllVerticesCount * entries[iEntry].vertices->getVertexCount();
-        boneIndicesAndWeights(iEntry,
-             indicesAndWeightsArray + pairSkipped,
-             pairsPerPrimitiveCount);
-        pairSkipped += pairsPerAllVerticesCount;
-    }
-    return *this;
-}
-
 RenderableManager::Builder& RenderableManager::Builder::boneIndicesAndWeights(size_t primitiveIndex,
-               const  utils::FixedCapacityVector< utils::FixedCapacityVector<filament::math::float2>> &indicesAndWeightsVector) {
-    auto count = indicesAndWeightsVector.size();
-    if (!count) {
-      // skip if no skinning data
-      // mImpl->mEntries[primitiveIndex].skinning.bonePairsSet = true;
-      return *this;
-    }
-    std::vector<Entry>& entries = mImpl->mEntries;
-    ASSERT_PRECONDITION(primitiveIndex < entries.size() && primitiveIndex >= 0,
-        "[primitive @ %u] index is out of size (%u)", primitiveIndex, entries.size());
-    auto vertexCount = entries[primitiveIndex].vertices->getVertexCount();
-    ASSERT_PRECONDITION(vertexCount > 0,
-        "[primitive @ %u] no defined vertices (%u)", primitiveIndex, vertexCount);
-    auto pairsPerVertexCount = count / vertexCount;
-    ASSERT_PRECONDITION(pairsPerVertexCount > 0 && count % vertexCount == 0,
-        "[primitive @ %u] bone indices and weights pairs count (%u) must be a multiple of vertex count (%u)",
-        primitiveIndex, count, vertexCount);
-    auto& skinning = entries[primitiveIndex].skinning;
-    ASSERT_PRECONDITION(skinning.bonePairs.empty(),
-        "[primitive @ %u] bone indices and weights data already exists", primitiveIndex);
-    skinning.bonePairs = indicesAndWeightsVector; // copy
-    skinning.bonePairsSet = true;
-
+               const  utils::FixedCapacityVector<
+                          utils::FixedCapacityVector<filament::math::float2>> &indicesAndWeightsVector) noexcept{
+    mImpl->mBonePairs[primitiveIndex] = indicesAndWeightsVector;
     return *this;
 }
 
 RenderableManager::Builder& RenderableManager::Builder::boneIndicesAndWeights(
-               const  utils::FixedCapacityVector<utils::FixedCapacityVector<utils::FixedCapacityVector<filament::math::float2>>> &indicesAndWeightsVectors) noexcept{
-  for (size_t iEntry = 0, c = min((uint)mImpl->mEntries.size(), indicesAndWeightsVectors.size()); iEntry < c; iEntry++)
+               const  utils::FixedCapacityVector<
+                          utils::FixedCapacityVector<
+                              utils::FixedCapacityVector<filament::math::float2>>> &indicesAndWeightsVectors) noexcept{
+  for (size_t iEntry = 0; iEntry < indicesAndWeightsVectors.size(); iEntry++)
```

Comment:
I think we should remove that API, because it's just a loop. The caller can do the same thing.

We can always add this back later, once the PR has landed if we think it makes thing a lot easier.

Reply:
Fixed